crates/ruff_python_formatter/src/pattern/pattern_match_class.rs

```diff
@@ -1,19 +1,41 @@
-use ruff_formatter::{write, Buffer, FormatResult};
+use ruff_formatter::{write, FormatResult};
 use ruff_python_ast::PatternMatchClass;
 
-use crate::{not_yet_implemented_custom_text, FormatNodeRule, PyFormatter};
+use crate::expression::parentheses::parenthesized;
+use crate::prelude::*;
+use crate::{FormatNodeRule, PyFormatter};
 
 #[derive(Default)]
 pub struct FormatPatternMatchClass;
 
 impl FormatNodeRule<PatternMatchClass> for FormatPatternMatchClass {
     fn fmt_fields(&self, item: &PatternMatchClass, f: &mut PyFormatter) -> FormatResult<()> {
-        write!(
-            f,
-            [not_yet_implemented_custom_text(
-                "NOT_YET_IMPLEMENTED_PatternMatchClass(0, 0)",
-                item
-            )]
-        )
+        let PatternMatchClass {
+            range,
+            cls,
+            patterns,
+            kwd_attrs,
+            kwd_patterns,
+        } = item;
+
+        let items = format_with(|f| {
+            let mut join = f.join_comma_separated(range.end());
+
+            if !patterns.is_empty() {
+                join.nodes(patterns.iter());
+            }
+
+            if !kwd_attrs.is_empty() {
+                for (key, value) in kwd_attrs.iter().zip(kwd_patterns.iter()) {
+                    join.entry(
+                        key,
+                        &format_with(|f| write!(f, [key.format(), text("="), value.format()])),
+                    );
+                }
+            }
+            join.finish()
+        });
+
+        write!(f, [cls.format(), parenthesized("(", &items, ")")])
```

Comment:
Try adding an extra group around items

Reply:
Would the function `group` work here?

Reply:
Yes, group(&items) should do the trick if I'm not mistaken

Reply:
Nothing changed for this `crates/ruff_python_formatter/tests/snapshots/black_compatibility@py_310__pattern_matching_extras.py.snap` file when using `group`

Reply:
See my comment below. This is fine, assuming we're talking about the same mismatch.